bundles/paikkatietoikkuna/coordinatetransformation/components/CoordinateSystemSelection.js

```diff
@@ -0,0 +1,241 @@
+                    '<div class="system coordinate" data-system="coordinate"><b class="dropdown_title"> ${ coordinate_system }</b> <a href="#"><div class="infolink icon-info"></div></a> <div class="select"></div>  </div> </br> ' +
+                    '<div class="system projection" style="display:none;" data-system="projection"> ${ map_projection } <a href="#"><div class="infolink icon-info"></div></a> <div class="select"></div> </div> </br>' +
+                    '<div class="system geodetic-coordinate" data-system="geodetic-coordinate"><b class="dropdown_title"> ${ geodetic_coordinate_system } *</b> <a href="#"><div class="infolink icon-info"></div></a> <div class="select"></div> </div> </br> ' +
+                    '<div class="system elevation" data-system="elevation"><b class="dropdown_title"> ${ elevation_system } </b> <a href="#"><div class="infolink icon-info"></div></a> <div class="select"></div> '+
+                '</div>'
+            ),
+        }
+        this.createUi();
+    }, {
+        getName: function() {
+            return 'Oskari.coordinatetransformation.view.SourceSelect';
+        },
+        setElement: function (el) {
+            this.element = el;
+        },
+        getElement: function () {
+            return this.element;
+        },
+        createUi: function () {
+            var wrapper = this._template.systemWrapper.clone();
+
+            var coordinateSystemSelection = this._template.coordinateSystemSelection({
+                title: this.loc.coordinatesystem.title,
+                geodetic_datum: this.loc.coordinatesystem.geodetic_datum,
+                coordinate_system: this.loc.coordinatesystem.coordinate_system,
+                map_projection: this.loc.coordinatesystem.map_projection,
+                geodetic_coordinate_system:this.loc.coordinatesystem.geodetic_coordinatesystem,
+                elevation_system:this.loc.coordinatesystem.heigth_system 
+            });
+            wrapper.append(coordinateSystemSelection);
+            this.setElement(wrapper);
+            this.populateSelect();
+            this.handleInfoLink();
+        },
+        /**
+         * @method populateSelect
+         * @desc creates an instance of the { Oskari.coordinatetransformation.component.select },
+         * and fills it with data
+         */
+        populateSelect: function () {
+            var me = this;
+            var wrapper = this.getElement();
+            this.select.create();
+            this.selectInstance = this.select.getSelectInstances();
+            this.dropdowns = this.select.getDropdowns();
+            Object.keys( this.dropdowns ).forEach( function( key ) {
+                jQuery( wrapper.find( '.transformation-system' ).find( me.makeClassSelector(key) ).append( me.dropdowns[key] ));
```

Comment:
Wouldn't it make more sense to bind the changelisteners here while creating the dropdowns instead of another function (handleSelectionChanged();) that finds new references for the dropdowns that are added here?

Reply:
Would this work?

    var dropdown = me.dropdowns[key];
    jQuery( wrapper.find( '.transformation-system' ).find( me.makeClassSelector(key) ).append( dropdown ));
    dropdown.on('change', function () {
        var value = jQuery(this).val();
        me.handleSelectValueChange(value);
    });

Then you could remove the next loop and the handleSystemChange-function.